lib/internal/wrap_js_stream.js

```diff
@@ -27,12 +30,15 @@ class JSStreamWrap extends Socket {
       debug('close');
       this.doClose(cb);
     };
-    handle.isAlive = () => this.isAlive();
-    handle.isClosing = () => this.isClosing();
-    handle.onreadstart = () => this.readStart();
-    handle.onreadstop = () => this.readStop();
-    handle.onshutdown = (req) => this.doShutdown(req);
-    handle.onwrite = (req, bufs) => this.doWrite(req, bufs);
+    // Inside of the following functions, `this` refers to the handle
+    // and `this.owner` refers to this JSStreamWrap instance.
+    handle.isClosing = function() { return this.owner.isClosing(); };
```

Comment:
would be slightly better to avoid the closures entirely and just make these top level functions.

Reply:
@jasnell done!